slither/detectors/shadowing/state.py

```diff
@@ -12,7 +12,7 @@ def detect_shadowing(contract: Contract):
     variables_fathers = []
     for father in contract.inheritance:
         if any(f.is_implemented for f in father.functions + father.modifiers):
-            variables_fathers += father.state_variables_declared
+            variables_fathers += [var for var in father.state_variables_declared if var.visibility != "private"]
 
     for var in contract.state_variables_declared:
```

Comment:
I think the heuristic should be updated to "are there any **public** state variables in the child contract that shadow **any** state variable in the parent". This would warn on instances where a variable is shadowed and they're not both private as the author suggested. Right now, this change no longer warns when a private variable is shadowed and there's a public one e.g. https://github.com/crytic/slither/blob/97b60d6115ec71264b9da4c63f60e92c5ae64c81/tests/detectors/shadowing-state/0.7.5/public_gap_variable.sol
```suggestion
            variables_fathers += father.state_variables_declared 

    for var in contract.state_variables_entry_points:
```
`state_variables_entry_points` is defined here https://github.com/crytic/slither/blob/master/slither/core/declarations/contract.py#L320

Reply:
@0xalpharush   @noyyyy what do think about my comment above?  how should i proceed?

Reply:
@montyly unless I'm mistaken, it appears the code wouldn't compile in cases where we would want this detector to to give a warning. and the only times it does compile and give the warning are cases where we want to suppress the warning